iTechArt.Repository/Repositories/RetirementRepository.cs

```diff
@@ -28,32 +28,105 @@ public RetirementRepository(AppDbContext dbContext, IMapper mapper)
-            int Y1985 = retiredPolices.Where(c => c.BirthDate.Year + RetirementAge >= 1985 && c.BirthDate.Year + RetirementAge < 1990).ToList().Count;
-            int Y1990 = retiredPolices.Where(c => c.BirthDate.Year + RetirementAge >= 1990 && c.BirthDate.Year + RetirementAge < 1995).ToList().Count;
-            int Y1995 = retiredPolices.Where(c => c.BirthDate.Year + RetirementAge >= 1995 && c.BirthDate.Year + RetirementAge < 2000).ToList().Count;
-            int Y2000 = retiredPolices.Where(c => c.BirthDate.Year + RetirementAge >= 2000 && c.BirthDate.Year + RetirementAge < 2005).ToList().Count;
-            int Y2005 = retiredPolices.Where(c => c.BirthDate.Year + RetirementAge >= 2005 && c.BirthDate.Year + RetirementAge < 2010).ToList().Count;
-            int Y2010 = retiredPolices.Where(c => c.BirthDate.Year + RetirementAge >= 2010 && c.BirthDate.Year + RetirementAge < 2015).ToList().Count;
-            int Y2015 = retiredPolices.Where(c => c.BirthDate.Year + RetirementAge >= 2015 && c.BirthDate.Year + RetirementAge < 2020).ToList().Count;
-            int Y2020 = retiredPolices.Where(c => c.BirthDate.Year + RetirementAge >= 2020 && c.BirthDate.Year + RetirementAge < 2022).ToList().Count;
-            int Y2022 = retiredPolices.Where(c => c.BirthDate.Year + RetirementAge == 2022).ToList().Count;
+            int Y1980to84 = retiredPolices.Where(c => c.BirthDate.Year+ RetirementAgePolice >= 1980 && c.BirthDate.Year+ RetirementAgePolice < 1985).ToList().Count;
+            int Y1985to89 = retiredPolices.Where(c => c.BirthDate.Year + RetirementAgePolice >= 1985 && c.BirthDate.Year + RetirementAgePolice < 1990).ToList().Count;
+            int Y1990to94 = retiredPolices.Where(c => c.BirthDate.Year + RetirementAgePolice >= 1990 && c.BirthDate.Year + RetirementAgePolice < 1995).ToList().Count;
+            int Y1995to99 = retiredPolices.Where(c => c.BirthDate.Year + RetirementAgePolice >= 1995 && c.BirthDate.Year + RetirementAgePolice < 2000).ToList().Count;
+            int Y2000to04 = retiredPolices.Where(c => c.BirthDate.Year + RetirementAgePolice >= 2000 && c.BirthDate.Year + RetirementAgePolice < 2005).ToList().Count;
+            int Y2005to09 = retiredPolices.Where(c => c.BirthDate.Year + RetirementAgePolice >= 2005 && c.BirthDate.Year + RetirementAgePolice < 2010).ToList().Count;
+            int Y2010to14 = retiredPolices.Where(c => c.BirthDate.Year + RetirementAgePolice >= 2010 && c.BirthDate.Year + RetirementAgePolice < 2015).ToList().Count;
+            int Y2015to19 = retiredPolices.Where(c => c.BirthDate.Year + RetirementAgePolice >= 2015 && c.BirthDate.Year + RetirementAgePolice < 2020).ToList().Count;
+            int Y2020to21 = retiredPolices.Where(c => c.BirthDate.Year + RetirementAgePolice >= 2020 && c.BirthDate.Year + RetirementAgePolice < 2022).ToList().Count;
+            int Y2022 = retiredPolices.Where(c => c.BirthDate.Year + RetirementAgePolice == 2022).ToList().Count;
```

Comment:
Filter must allow user to set custom date range

Reply:
It is nor filter, it is for graph data